llama_stack/distribution/datatypes.py

```diff
@@ -134,6 +135,45 @@ class ServerConfig(BaseModel):
     )
 
 
+class UserConfig(BaseModel):
```

Comment:
not very clear we need UserConfig. There's already a SecretStr type so that we dont return secrets

Reply:
@raghotham yeah, I could use that. I was thinking `UserConfig` could be useful for really explicitly choosing what parts of the configuration we want to expose to the user rather than showing everything and just hiding parts with `SecretStr`

Reply:
Can we add UserConfig as a next step? For now, expose everything other than SecretStr?

Reply:
sure yeah, fine by me!

Reply:
changed